### PR TITLE
Homme/preqx kokkos summit mods1

### PR DESCRIPTION
**Pull request by ambrad**
This PR does the following:
- Fix a compile error in the code meant to handle the case that MPI does not handle on-device pointers.
- Work around a Cuda 9.1.* parse error in this same section of code.
- Fix some deprecated Kokkos usage.
- Fix a bug in Homme in which accum%IEner_wet was not being 0-initialized, leading to NaNs in repro sum in preqx_kokkos.

[BFB]

### CODE REVIEW

**Comment by mt5555** [comment]
looks like the %accum initialization code should move into the model specific init code.  but no reason to do that now  in this PR, I'll put it on my todo list.

**Comment by ambrad** [comment]
@mt5555, I don't mind implementing that approach in this PR if you'd like. Do you agree that I should modify
```
    call prim_printstate_init(par)
```
which is called in prim_init1_elem_arrays, to instead be
Do you agree that I should modify
```
    call prim_printstate_init(par, elem)
```
and then model-specific prim_state_mod impls can 0-init the %accum variable relevant to prim_state_mod?

**Comment by mt5555** [comment]
@ambrad - moving to prim_printstate_init() seems ideal, thanks.

File: components/homme/src/share/cxx/utilities/SubviewUtils.hpp [hunk 1]
```diff
@@ -26,7 +26,7 @@ subview(ViewType<ScalarType * [DIM1], MemSpace, Properties...> v_in, int ie) {
   assert(ie < v_in.extent_int(0));
   assert(ie >= 0);
   return ViewUnmanaged<ScalarType[DIM1], MemSpace>(
-      &v_in.implementation_map().reference(ie, 0));
+      &v_in.impl_map().reference(ie, 0));
```
Comment (oksanaguba) [inline components/homme/src/share/cxx/utilities/SubviewUtils.hpp]: You changed name of member, so this code is/was never built?

Comment (bartgol) [reply components/homme/src/share/cxx/utilities/SubviewUtils.hpp]: I think kokkos currently allows both. The old one would still compile now, but it won't sometime in the future.

Comment (bartgol) [reply components/homme/src/share/cxx/utilities/SubviewUtils.hpp]: This code is built. We use it every time we take a subview in HOMMEXX.

Comment (ambrad) [reply components/homme/src/share/cxx/utilities/SubviewUtils.hpp]: Right. imlementation_map is deprecated and impl_map is the replacement. Currently both work unless you build Kokkos with deprecated code disallowed.

**Comment by ambrad** [comment]
@jgfouca I'm going to implement the other approach Mark and I discussed above, so don't merge this yet. I'll change the title to include [WIP] until that's done.

**Comment by ambrad** [comment]
@mt5555, I made the change. While I was at it, I also fixed a compiler warning.

**Comment by jgfouca** [comment]
Merged to next.

**Pull request merged**